File: daily/Arrays/832.Flipping-an-image.py
from typing import List


class Solution:
    def flipAndInvertImage(self, image: List[List[int]]) -> List[List[int]]:
        rows = len(image)
        cols = len(image[0]) if rows else 0
        for i in range(rows):
            left, right = 0, cols - 1
            while left <= right:
                if image[i][left] == image[i][right]:
                    image[i][left] ^= 1
                    image[i][right] = image[i][left]
                left += 1
                right -= 1
        return image

    def flipAndInvertImageV2(self, image: List[List[int]]) -> List[List[int]]:
        rows = len(image)
        cols = len(image[0]) if rows else 0
        for i in range(rows):
            left, right = 0, cols - 1
            while left <= right:
                image[i][left], image[i][right] = image[i][right], image[i][left]
                left += 1
                right -= 1
        for i in range(rows):
            for j in range(cols):
                image[i][j] ^= 1
        return image

File: daily/Arrays/832.Flipping-an-image.go
package daily

func flipAndInvertImageV2(image [][]int) [][]int {
	rows := len(image)
	columns := len(image[0])

	for i := 0; i < rows; i++ {
		left := 0
		right := columns - 1

		for left <= right {
			image[i][left], image[i][right] = image[i][right], image[i][left]
			left++
			right--
		}
	}

	for i := 0; i < rows; i++ {
		for j := 0; j < columns; j++ {
			switch image[i][j] {
			case 0:
				image[i][j] = 1
			case 1:
				image[i][j] = 0
			}
		}
	}

	return image
}

func flipAndInvertImage(image [][]int) [][]int {
	rows := len(image)
	columns := len(image[0])

	for i := 0; i < rows; i++ {
		left, right := 0, columns-1
		for left <= right {
			if image[i][left] == image[i][right] {
				image[i][left] ^= 1
				image[i][right] = image[i][left]
			}
			left++
			right--
		}
	}

	return image
}
